lib/SILGen/SILGenApply.cpp

```diff
@@ -5630,6 +5626,12 @@ RValue SILGenFunction::emitApply(
         });
   }
 
+  if (!calleeTypeInfo.foreign.async) {
+    // For a non-foreign-async callee, we hop back to the current executor
```

Comment:
@kavon on main can you expand this comment so that it explains what goes wrong if we insert it before popping the argument scope. This is a "what" comment instead of a "why" comment.